amivapi/auth/auth.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+#
+# license: AGPLv3, see LICENSE for details. In addition we strongly encourage
+#          you to buy us beer if we meet and you like the software.
+
+"""Custom AMIV Token auth class.
+
+**FAQ**
+
+    # access authentication data of current user
+    from flask import g
+    g.get('current_user')       # user_id (str)
+    g.get('current_token')      # token (str)
+    g.get('current_session')    # session data (dict)
+
+    # Check if the user is admin
+    from flask import g
+    if g.get('resource_admin'):
+        print("Can see and change everything!")
+    if g.get('resource_admin_readonly'):
+        print('Can look at anything, but not necessarily change something.')
+
+    # Add additinonal authentication with a new hook
+    def my_auth_hook(resource):
+        # You could set g.resource_admin or g.resource_admin_readonly here
+        pass
+    app.after_auth += my_auth_hook
+
+
+**How to use**
+
+1. Subclass `AmivTokenAuth`
+2. implement `create_user_lookup_filter`, `has_item_write_permission` and/or
+   'has_resource_write_permission' if you don't want default behaviour.
+   (Default: Users can't write, no special lookups)
+   Note: You shouldn't care about admin permissions. Those methods will only be
+   called for non-admins!
+3. Use your auth class in your resource settings. Done!
+4. Tip: You can still use all Eve settings like 'public_methods' and
+   'public_item_methods'!
+
+Take a look at `users.security` for an example.
+
+
+**How it works**
+
+We are using Eve's hooks to add authentiation.
+All methods go through the following methods:
```

Comment:
Indeed. (Except GET on home,  which is handled a bit special by eve,  but all requests to api resources and items.)